### DISCUSSION TITLE
Should CellError ane the ErrorMessage be exported?

### DISCUSSION THREAD

**Discussion by MartinDawson** [handsontable/hyperformula#734] [General]
I have a custom plugin like so:

```js
import { FunctionPlugin, DetailedCellError, ErrorType } from "hyperformula";

export class FinancialPlugin extends FunctionPlugin {
  fin({ args }, cellAddress) {
    if (!args.length) {
      const error = new CellError(ErrorType.NA, ErrorMessage.WrongArgNumber);
      const detailedCellError = new DetailedCellError(error, error.type, cellAddress);
      
      return detailedCellError;
    }
  }
}

FinancialPlugin.implementedFunctions = {
  FIN: {
    method: "fin",
  },
};

export const finTranslations = {
  enGB: {
    FIN: "FIN",
  },
};
```

I see this PR: https://github.com/handsontable/hyperformula/pull/369/commits/a64c43f374ebe61b8276a6adcec2acb3d3195739

And it added DetailedCellError export but not CellError.

I basically want to return a DetailedCellError if the user hasn't typed in enough arguments into my custom function. I'd like to keep in line with hyperformula existing error structure for this.

How can I do this if the CellError is not exported from hyperformula?

I see their is also this error type too: `InvalidArgumentsError();`. So maybe I should be using this instead but this returns a string and not the object with other information like: 'N/A' or '!REF' that I was looking for.

Thanks

**Comment by MartinDawson** [comment]
To be honest I guess we don't really need the other information like cellAddress and type like '!REF' which is why they aren't exported and I should just use the InvalidArgumentsError

**Comment by izulin** [comment]
I believe this is an oversight on our side and `CellError` should be exported.

**Reply by wojciechczerniak** [reply]
For Plugin authors right? And the wrong assumption was that only we use it currently so it is internal thing

**Comment by wojciechczerniak** [comment]
@MartinDawson `DetailedCellError` is created from `CellError`, it adds translation and the location. Functions and plugins should return `CellError`, we transform it automatically.

@izulin correct me if I'm wrong. So we would need to export `CellError` for plugin authors or merge those two. Also @kirszenbaum is going to write more on custom functions in #557 and this question is a material for a section at least.
